Source/WebCore/html/HTMLLabelElement.cpp

```diff
@@ -64,16 +65,16 @@ Ref<HTMLLabelElement> HTMLLabelElement::create(const QualifiedName& tagName, Doc
     return adoptRef(*new HTMLLabelElement(tagName, document));
 }
 
-RefPtr<LabelableElement> HTMLLabelElement::control() const
+RefPtr<HTMLElement> HTMLLabelElement::control() const
 {
     auto& controlId = attributeWithoutSynchronization(forAttr);
     if (controlId.isNull()) {
         // Search the children and descendants of the label element for a form element.
         // per http://dev.w3.org/html5/spec/Overview.html#the-label-element
         // the form element must be "labelable form-associated element".
-        for (auto& labelableElement : descendantsOfType<LabelableElement>(*this)) {
-            if (labelableElement.supportLabels())
-                return const_cast<LabelableElement*>(&labelableElement);
+        for (const auto& labelableElement : descendantsOfType<HTMLElement>(*this)) {
+            if (labelableElement.isLabelable())
+                return const_cast<HTMLElement*>(&labelableElement);
```

Comment:
It seems like this static cast is unnecessary?

Reply:
Unfortunately it is: I've tried removing it and it doesn't compile, because this is a `const` function, meaning `*this` gets us a `const` pointer, so the type of `descendantsOfType` range item is `const HTMLElement`, which we can cast only to `RefPtr<const HTMLElement>`, which doesn't work with WebIDL bindings conversions.